### ISSUE TITLE
Abstract it away from mongodb

### ISSUE THREAD

**Issue by masaeedu** [mongodb-js/mongodb-schema#48]
This tool would be pretty handy for getting some initial schema information on a large blob of semistructured JSON (or JSON-convertible) data. It doesn't seem like there's anything about the principle of operation that necessarily ties it to mongo, so maybe some part can be factored out that makes it work for arbitrary JSON data.

**Comment by pzrq** [comment]
@masaeedu Could you describe your specific use case, user story or motivation for suggesting this in more depth?

Asides: I'm on holidays at the moment but occasionally working on side projects (and full disclaimer I work for MongoDB).

Little value in building it without more details, PRs are of course always welcome 😄

**Comment by masaeedu** [comment]
@pzrq Sure. I have a large XML DSL codebase that I just converted to JSON. There isn't a 1-1 correspondence between concepts in XML and in JSON (e.g. there's no such thing as "tagnames"), so the conversion simplifies some stuff and is lossy. This means there isn't a simple mechanical way to port the old code that deserialized and worked with the DSL to the new format.

In order to understand the resulting configuration set, it would be better to mechanically infer a schema from the entirety of the schema set and write code targeting that, rather than trying to browse the data ad-hoc hoping to spot all differences of interest.

TL;DR I want something like SlamData-lite for large sets of raw JSON data.

**Comment by pzrq** [comment]
@masaeedu Sounds like something someone might be willing to pay a lot of 💰 for 😉 

In the absence of guidance from others like @rueckstiess who is on vacation at this time too, I can suggest you review the https://github.com/mongodb-js/mongodb-schema/blob/master/LICENSE and decide if forking this project to have a better starting point is appropriate for your needs.

**Comment by pzrq** [comment]
Closing due to [inactivity](https://github.com/mongodb-js/mongodb-schema/network/members) at this time. 

@masaeedu @rueckstiess
Please feel welcome to reopen if there is any work planned and you'd like to use this to track its progress.
